### PR TITLE
tasks completed

### PR DESCRIPTION
**Pull request by BaLidor**
first task:
changes to identity.sol and IdentityGuard.sol

second task:
new folder called 'lidor'.
under this folder you can find changes in the schema and mappings files.

### CODE REVIEW

File: .idea/GoodContracts.iml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sirpy) [inline .idea/GoodContracts.iml:1]: dont upload your .idea files:)

File: contracts/identity/Identity.sol [hunk 2]
```diff
@@ -294,3 +294,374 @@ contract Identity is IdentityAdminRole, SchemeGuard, Pausable {
...
+        return (identity.isContract(_addr) || _isContract2(_addr));
+    }
+
+    /* @dev Function to update DID, only by the account owner
+     * @param account the address of the account owner
+     * @param did the new DID to update
+     */
+    function updateDID(address account, string memory did)
+      public
+      onlyAccountOwner
```
Comment (sirpy) [inline contracts/identity/Identity.sol:653]: this is missing something

File: lidor/src/mapping.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+export function handleAuthenticate(call: AuthenticateCall): void {
+  let entity = ExampleEntity.load(call.transaction.from.toHex())
+  if (entity == null) {
+    entity = new ExampleEntity(call.transaction.from.toHex())
+    entity.count = BigInt.fromI32(0)
+  }
+  entity.count = entity.count + BigInt.fromI32(1)
+  entity.account = call.params.account
+  entity.save()
```
Comment (sirpy) [inline lidor/src/mapping.ts:113]: what is this entity?

File: lidor/src/mapping.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+  let citizen = Citizen.load(call.transcation.from.toHex())
```
Comment (sirpy) [inline lidor/src/mapping.ts:115]: will this give you the citizen record you are looking for?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: lidor/src/mapping.ts (hunk 3 above)
Comment (BaLidor) [reply lidor/src/mapping.ts:113]: I wasn't supposed to submit this one, I put it there for me as an example from the other entity and forgot to remove it, it should have been like this:

export function handleAuthenticate(call: AuthenticateCall): void {

  let citizen = Citizen.load(call.transcation.from.toHex())
  if (citizen == null) {
    citizen = new Citizen(call.transaction.from.toHex())
  }
  citizen.account = call.params.account
  citizen.lastAuthenticated = call.block.timestamp
  citizen.save()
}

File: lidor/src/mapping.ts (hunk 4 above)
Comment (BaLidor) [reply lidor/src/mapping.ts:115]: I am not sure of the purpose of the citizen record - I found the documentation of the etherium.call class so I am aware of the existence of call.to / call.from / call.transaction.to... but I am not sure as to which one exactly is required for the citizen entity. (is the citizen the one initiating the call?)

File: lidor/src/mapping.ts (hunk 3 above)
Comment (BaLidor) [reply lidor/src/mapping.ts:113]: this was mistakenly submitted, I used this for myself as an example but apparently forgot to delete this part, it should be:

export function handleAuthenticate(call: AuthenticateCall): void {

  let citizen = Citizen.load(call.transcation.from.toHex())
  if (citizen == null) {
    citizen = new Citizen(call.transaction.from.toHex())
  }
  citizen.account = call.params.account
  citizen.lastAuthenticated = call.block.timestamp
  citizen.save()
}

File: contracts/identity/Identity.sol (hunk 2 above)
Comment (BaLidor) [reply contracts/identity/Identity.sol:653]: I'm missing the parameter for the modifier, it should be:

function updateDID(address account, string memory did)
      public
      onlyAccountOwner(account)

File: lidor/src/mapping.ts (hunk 4 above)
Comment (sirpy) [reply lidor/src/mapping.ts:115]: if you are listening to `authenticate` calls where is there identifier that will give you the citizen record you need.
what do you mean you are not sure what is the citizen record for? that's the record that stores the latest authentication status of the account ie user ie citizen

Comment (BaLidor) [reply lidor/src/mapping.ts:115]: then I assume the wanted address is the one given in the input of the authenticate function:

  let citizen = Citizen.load(call.inputs.account)
  if (citizen == null) {
    citizen = new Citizen(call.inputs.account)
  }

**Pull request closed without merging**